src/cr/cube/dimension.py

```diff
@@ -650,6 +672,25 @@ def _iter_element_makings(self):
             yield idx, element_dict, element_transforms_dict
 
 
+class _DerivedElements(_BaseElements):
+    """Sequence of derived element objects for a dimension.
+
+    *derived_elements* is a sequence of _Elements containing the elements
+    of a dimension that are derived (aka zz9 computed insertions on array variables).
+    This object is only intended to be constructed by _AllElements.derived_elements and
+    there should be no reason to construct it directly.
+    """
+
+    def __init__(self, all_elements, dimension_transforms_dict):
+        self._all_elements = all_elements
+        self._dimension_transforms_dict = dimension_transforms_dict
```

Comment:
Why is this needed in this class?

Reply:
Hm, on closer look I'm not sure why. I was copying from `_ValidElements` which also has it, but I don't see anything different. Will investigate

Reply:
@gergness just a note on `_ValidElements` and possibly applicable to `_DerivedElements`. The reason for a "subset" subclass is essentially to make the collection and special properties and methods defined in `_BaseElements` available on these subset classes. The collection and special properties/methods are:

* indexed access: `elements[idx]` or `elements[n:m]`
* iteration: `for e in elements`
* len: `len(elements)`
* .element_ids: `elements.element_ids -> (1, 4, 5)`
* .element_idxs: `elements.element_idxs -> (0, 3, 4)`
* .get_by_id(element_id): `elements.get_by_id(1) -> elements[0]`

All of these methods are computation on the `._elements` instance variable, so changing that to be a subset provides all these methods for "free" in a subclass. A subclass is also free to define new methods specific to its subset if appropriate.

You could possibly return a tuple of Element objects instead if you knew for sure you didn't need more than the sequence methods (indexed access, iteration, and len) that could just as well be provided by `tuple()`. I'd be inclined to think subclassing is a better bet though so you don't need to compute element-ids or payload-order element-idxs yourself; those seem likely to be needed by someone somewhere along the line. Maybe lookups too if you need to find a derived element for sorting purposes.